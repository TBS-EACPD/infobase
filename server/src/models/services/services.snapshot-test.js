const services_fields = `
services {
  service_id
  org_id
  org {
    name
  }
  program_ids
  programs {
    name
  }

  name
  description
  service_type
  scope
  target_groups
  feedback_channels
  urls

  last_gender_analysis

  collects_fees
  account_reg_digital_status
  authentication_status
  application_digital_status
  decision_digital_status
  issuance_digital_status
  issue_res_digital_status
  digital_enablement_comment

  service_report {
    service_id
    year
    cra_business_ids_collected
    SIN_collected
    phone_inquiry_count
    online_inquiry_count
    online_application_count
    live_application_count
    mail_application_count
    other_application_count
    service_report_comment
  }

  standards {
    standard_id,
    service_id,

    name
    last_gcss_tool_year
    channel
    type
    other_type_comment

    target_type

    urls
    rtp_urls

    standard_report {
      standard_id,
      year
      lower
      count
      met_count
      standard_report_comment  
    }
  },
}
`;

const all_services_and_standards_for_org = `
query{
  root(lang: "en"){
    org(org_id: "326"){
      ${services_fields}
    }
  }
}`;

const all_services_and_standards_for_program = `
query{
  root(lang: "en"){
    program(id: "TBC-BXB01"){
      ${services_fields}
    }
  }
}`;

const { execQuery } = global;

describe("services data", function () {
  it("All services and standards for org", async () => {
    const data = await execQuery(all_services_and_standards_for_org, {});
    return expect(data).toMatchSnapshot();
  });
  it("All services and standards for program", async () => {
    const data = await execQuery(all_services_and_standards_for_program, {});
    return expect(data).toMatchSnapshot();
  });
});
