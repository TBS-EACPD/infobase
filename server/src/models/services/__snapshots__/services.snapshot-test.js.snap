// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services data All services and standards for org 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "services": Array [
          Object {
            "account_reg_digital_status": false,
            "application_digital_status": true,
            "authentication_status": false,
            "collects_fees": false,
            "decision_digital_status": null,
            "description": "This service enables institutions subject to the Access to Information Act and the Privacy Act to receive and respond to Access to Information and Privacy (ATIP) requests, including fees collected electronically (where applicable).",
            "digital_enablement_comment": "This service does not use the decision interaction point.",
            "feedback_channels": Array [
              "Telephone",
              "Email",
              "Postal Mail",
              "Online",
            ],
            "issuance_digital_status": true,
            "issue_res_digital_status": true,
            "last_gender_analysis": "2017",
            "name": "ATIP Online Request Service",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-ISS07",
            ],
            "programs": Array [
              Object {
                "name": "Information Technology Services - Treasury Board Secretariat",
              },
            ],
            "scope": Array [
              "Internal Enterprise Service",
            ],
            "service_id": "1297",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": null,
                "online_application_count": 3289,
                "online_inquiry_count": null,
                "other_application_count": null,
                "phone_inquiry_count": null,
                "service_id": "1297",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Information Technology Services",
              "Regulatory Compliance and Enforcement",
            ],
            "standards": Array [],
            "target_groups": Array [
              "Internal to Government",
            ],
            "urls": Array [
              "https://www.canada.ca/en/treasury-board-secretariat/services/access-information-privacy/access-information/request-information.html",
            ],
          },
          Object {
            "account_reg_digital_status": null,
            "application_digital_status": null,
            "authentication_status": null,
            "collects_fees": false,
            "decision_digital_status": null,
            "description": "Answers queries from the general public, as well as federal government employees.",
            "digital_enablement_comment": "This service does not use these client interaction points.",
            "feedback_channels": Array [
              "Other",
            ],
            "issuance_digital_status": null,
            "issue_res_digital_status": null,
            "last_gender_analysis": null,
            "name": "Public Enquiries",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-ISS02",
            ],
            "programs": Array [
              Object {
                "name": "Communications Services - Treasury Board Secretariat",
              },
            ],
            "scope": Array [
              "External Service",
            ],
            "service_id": "1114",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": 17,
                "online_application_count": null,
                "online_inquiry_count": null,
                "other_application_count": 4731,
                "phone_inquiry_count": 611,
                "service_id": "1114",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Communications Services",
            ],
            "standards": Array [],
            "target_groups": Array [
              "Economic Segments",
              "Environmental",
              "Foreign Entities",
              "Non-Profit Institutions and Organizations",
              "Persons",
              "Provinces, Territories and Communities",
            ],
            "urls": Array [
              "https://www.canada.ca/en/treasury-board-secretariat.html",
              "https://www.canada.ca/en/treasury-board-secretariat/corporate/contact.html",
              "https://open.canada.ca/en/forms/contact-us",
            ],
          },
          Object {
            "account_reg_digital_status": null,
            "application_digital_status": true,
            "authentication_status": true,
            "collects_fees": false,
            "decision_digital_status": true,
            "description": "This service is in place to compensate employees who have incurred expenses or financial losses because of the implementation of the Phoenix pay system.",
            "digital_enablement_comment": "This service does not use online account registration and enrolment.",
            "feedback_channels": Array [
              "Telephone",
              "Email",
            ],
            "issuance_digital_status": true,
            "issue_res_digital_status": true,
            "last_gender_analysis": null,
            "name": "Treasury Board of Canada Secretariatâ€™s Claims Office",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-BXB01",
            ],
            "programs": Array [
              Object {
                "name": "Financial Management Policies and Initiatives",
              },
            ],
            "scope": Array [
              "Internal Enterprise Service",
            ],
            "service_id": "1348",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": null,
                "online_application_count": 2014,
                "online_inquiry_count": null,
                "other_application_count": null,
                "phone_inquiry_count": null,
                "service_id": "1348",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Advisory Services",
            ],
            "standards": Array [
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Acknowledge the receipt of all claims within one business day.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1361",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1361",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Provide a recommendation within 10 business days.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1577",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1577",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Provide a recommendation within 60 business days for claims related to income taxes and benefits.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1578",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1578",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Provide a recommendation within 60 business days for very complex claims.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1579",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1579",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
            ],
            "target_groups": Array [
              "Internal to Government",
              "Persons",
            ],
            "urls": Array [
              "https://www.canada.ca/en/treasury-board-secretariat/topics/pay/phoenix-pay-system/phoenix.html",
            ],
          },
          Object {
            "account_reg_digital_status": true,
            "application_digital_status": true,
            "authentication_status": false,
            "collects_fees": true,
            "decision_digital_status": false,
            "description": "The Access to Information Act gives Canadian citizens, permanent residents, and any person or corporation present in Canada a right to access records of government institutions that are subject to the Act. The Privacy Act gives Canadian citizens, permanent residents, and individuals present in Canada the right to access their personal information held by government institutions that are subject to the Act, and protects that information against unauthorized collection, use, retention and disclosure. This service responds to Access to Information and Privacy requests received by the Treasury Board Secretariat (TBS).",
            "digital_enablement_comment": null,
            "feedback_channels": Array [
              "Telephone",
              "Email",
              "Postal Mail",
              "Online",
              "Other",
            ],
            "issuance_digital_status": false,
            "issue_res_digital_status": false,
            "last_gender_analysis": null,
            "name": "Access to Information and Privacy",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-ISS01",
            ],
            "programs": Array [
              Object {
                "name": "Management and Oversight Services - Treasury Board Secretariat",
              },
            ],
            "scope": Array [
              "External Service",
            ],
            "service_id": "1347",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": 0,
                "mail_application_count": 69,
                "online_application_count": 572,
                "online_inquiry_count": null,
                "other_application_count": 0,
                "phone_inquiry_count": null,
                "service_id": "1347",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Advisory Services",
            ],
            "standards": Array [
              Object {
                "channel": null,
                "last_gcss_tool_year": null,
                "name": "The legislated timeframe for responding to an access to information request or a request for personal information is 30 days.",
                "other_type_comment": "This service standard is administered by mail and email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1347",
                "standard_id": "1359",
                "standard_report": Array [
                  Object {
                    "count": 594,
                    "lower": 90,
                    "met_count": 575,
                    "standard_id": "1359",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
            ],
            "target_groups": Array [
              "Economic Segments",
              "Environmental",
              "Internal to Government",
              "Non-Profit Institutions and Organizations",
              "Persons",
              "Provinces, Territories and Communities",
            ],
            "urls": Array [
              "https://atip-aiprp.apps.gc.ca/atip/welcome.do",
            ],
          },
          Object {
            "account_reg_digital_status": true,
            "application_digital_status": true,
            "authentication_status": true,
            "collects_fees": false,
            "decision_digital_status": null,
            "description": "Provide human resources data engineering and analytics services for corporate reporting and employer accountabilities",
            "digital_enablement_comment": "This service is internal to government. For online requests, clients must have a Government of Canada email account, and must register online. No monetary payments are required from clients. At any point in time, client may also email a generic mailbox for any questions they may have regarding their requests and/or making requests in general. This service does not use online decision and issuance functions.",
            "feedback_channels": Array [
              "Telephone",
              "In-Person",
              "Email",
              "Online",
            ],
            "issuance_digital_status": null,
            "issue_res_digital_status": true,
            "last_gender_analysis": null,
            "name": "People Management Information/Data Request Service",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-BXC03,ISS04",
            ],
            "programs": Array [
              null,
            ],
            "scope": Array [
              "Internal Enterprise Service",
            ],
            "service_id": "1094",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": null,
                "online_application_count": 420,
                "online_inquiry_count": null,
                "other_application_count": 0,
                "phone_inquiry_count": null,
                "service_id": "1094",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Management and Oversight Services",
              "Human Resources Management",
            ],
            "standards": Array [],
            "target_groups": Array [
              "Internal to Government",
            ],
            "urls": Array [
              "https://publiservice.ochro-bdprh.tbs-sct.gc.ca/pimart-oragip/home-eng.aspx",
            ],
          },
          Object {
            "account_reg_digital_status": true,
            "application_digital_status": null,
            "authentication_status": null,
            "collects_fees": false,
            "decision_digital_status": null,
            "description": "The Open Government Portal provides a mechanism for members of the public to request data and information to be included in the Open Government Portal.",
            "digital_enablement_comment": "This service does not use online authentication, application, decision or issuance functions.",
            "feedback_channels": Array [
              "Email",
            ],
            "issuance_digital_status": null,
            "issue_res_digital_status": true,
            "last_gender_analysis": null,
            "name": "Open Government Database and Information Requests",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-ISS07",
            ],
            "programs": Array [
              Object {
                "name": "Information Technology Services - Treasury Board Secretariat",
              },
            ],
            "scope": Array [
              "External Service",
            ],
            "service_id": "1349",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": null,
                "online_application_count": 92,
                "online_inquiry_count": null,
                "other_application_count": null,
                "phone_inquiry_count": null,
                "service_id": "1349",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Resources",
            ],
            "standards": Array [],
            "target_groups": Array [
              "Environmental",
              "Internal to Government",
              "Non-Profit Institutions and Organizations",
              "Persons",
              "Provinces, Territories and Communities",
            ],
            "urls": Array [
              "https://open.canada.ca/en/forms/suggest-dataset",
            ],
          },
          Object {
            "account_reg_digital_status": null,
            "application_digital_status": null,
            "authentication_status": null,
            "collects_fees": false,
            "decision_digital_status": null,
            "description": "The Office of the Chief Human Resources Officer (OCHRO) issues accreditation to classification advisors in the Core Public Administration (CPA) who have completed the Organization and Classification (OC) learning program as stated in the OC learning curriculum and in accordance with section 6.2.3 of the Policy on Classification.   

The classification accreditation responds to the requirement of the Policy on Classification that allows the Deputy Head to sub-delegate the authority to make classification decision only to accredited advisors. The OCHRO provides advice on the assessment of accreditation applications and assesses the applications once completed.",
            "digital_enablement_comment": "The following client interaction points are available through email or online (GConnex): requests for advice and guidance; accreditation submissions; follow-up questions and actions; and issuance of accreditation certificates.",
            "feedback_channels": Array [
              "Telephone",
              "Email",
              "Postal Mail",
              "Online",
            ],
            "issuance_digital_status": null,
            "issue_res_digital_status": null,
            "last_gender_analysis": null,
            "name": "Classification Accreditation",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-BXC01,ISS04",
            ],
            "programs": Array [
              null,
            ],
            "scope": Array [
              "Internal Enterprise Service",
            ],
            "service_id": "1350",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": 8,
                "online_application_count": 1,
                "online_inquiry_count": null,
                "other_application_count": 4,
                "phone_inquiry_count": null,
                "service_id": "1350",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Human Resources Management",
            ],
            "standards": Array [],
            "target_groups": Array [
              "Internal to Government",
            ],
            "urls": Array [
              "",
            ],
          },
        ],
      },
    },
  },
}
`;

exports[`services data All services and standards for program 1`] = `
Object {
  "data": Object {
    "root": Object {
      "program": Object {
        "services": Array [
          Object {
            "account_reg_digital_status": null,
            "application_digital_status": true,
            "authentication_status": true,
            "collects_fees": false,
            "decision_digital_status": true,
            "description": "This service is in place to compensate employees who have incurred expenses or financial losses because of the implementation of the Phoenix pay system.",
            "digital_enablement_comment": "This service does not use online account registration and enrolment.",
            "feedback_channels": Array [
              "Telephone",
              "Email",
            ],
            "issuance_digital_status": true,
            "issue_res_digital_status": true,
            "last_gender_analysis": null,
            "name": "Treasury Board of Canada Secretariatâ€™s Claims Office",
            "org": Object {
              "name": "Treasury Board of Canada Secretariat",
            },
            "org_id": "326",
            "program_ids": Array [
              "TBC-BXB01",
            ],
            "programs": Array [
              Object {
                "name": "Financial Management Policies and Initiatives",
              },
            ],
            "scope": Array [
              "Internal Enterprise Service",
            ],
            "service_id": "1348",
            "service_report": Array [
              Object {
                "SIN_collected": null,
                "cra_business_ids_collected": null,
                "live_application_count": null,
                "mail_application_count": null,
                "online_application_count": 2014,
                "online_inquiry_count": null,
                "other_application_count": null,
                "phone_inquiry_count": null,
                "service_id": "1348",
                "service_report_comment": null,
                "year": "2018",
              },
            ],
            "service_type": Array [
              "Advisory Services",
            ],
            "standards": Array [
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Acknowledge the receipt of all claims within one business day.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1361",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1361",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Provide a recommendation within 10 business days.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1577",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1577",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Provide a recommendation within 60 business days for claims related to income taxes and benefits.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1578",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1578",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
              Object {
                "channel": null,
                "last_gcss_tool_year": "2017",
                "name": "Provide a recommendation within 60 business days for very complex claims.",
                "other_type_comment": "The service standard is administered by email.",
                "rtp_urls": Array [
                  "",
                ],
                "service_id": "1348",
                "standard_id": "1579",
                "standard_report": Array [
                  Object {
                    "count": 0,
                    "lower": 90,
                    "met_count": 0,
                    "standard_id": "1579",
                    "standard_report_comment": null,
                    "year": "2018",
                  },
                ],
                "standard_type": null,
                "target_type": "Percentage of outcomes",
                "urls": null,
              },
            ],
            "target_groups": Array [
              "Internal to Government",
              "Persons",
            ],
            "urls": Array [
              "https://www.canada.ca/en/treasury-board-secretariat/topics/pay/phoenix-pay-system/phoenix.html",
            ],
          },
        ],
      },
    },
  },
}
`;
